lib/application/widgets/theme/tags_field.dart

```diff
@@ -0,0 +1,273 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/animations.dart' as anims;
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/images.dart' as images;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/hooks/tags_controller_hook.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+
+/// Provides the list of tags of the collection being updated.
+///
+/// These tags may be mutated by [TagsField], which the user may use to update the tags list.
+class TagsController extends ValueNotifier<List<String>> {
+  TagsController({List<String>? tags}) : super(tags ?? []);
+
+  List<String> get tags => value;
+
+  set tags(List<String> tags) => value = List.from(tags);
+}
+
+/// A custom [TextField] that groups a collection of tags.
+class TagsField extends HookConsumerWidget {
+  const TagsField({this.controller, this.maxTags = 5, this.errorText});
+
+  /// Controls the tags being edited.
+  ///
+  /// If null, this widget will create its own [TagsController].
+  final TagsController? controller;
+
+  /// {@template TagsDropdownField.maxTags}
+  /// Maximum number of tags allowed to be chosen.
+  ///
+  /// Inputs that would increase this amount will be ignored, i.e, not transformed in tags.
+  /// {@endtemplate}
+  final int maxTags;
+
+  /// {@macro CustomTextField.errorText}
+  final String? errorText;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
```

Comment:
This widget kind-of looks too big, I think it could be split into one or multiple smaller widget(s). This would benefit in being more performant on updates, as well all of being specific on its responsibility, granting more "succinctness" to each widget/nested widget.

Reply:
I moved the `tagsFieldContainer` to a separate Widget which improved a little bit the main Widget size. I tried some different approaches trying to shrink the `useEffect` size but the problem is that the 3 controllers (1. tags controller 2. text field controller and 3. focus controller) are very coupled because the act of selecting a tag impact in all of them. Let me know what do you think.